controllers/smbshare_controller.go

```diff
@@ -72,9 +73,7 @@ func (r *SmbShareReconciler) Reconcile(
 }
 
 func (r *SmbShareReconciler) setRecorder(mgr ctrl.Manager) {
-	if r.recorder == nil {
-		r.recorder = mgr.GetEventRecorderFor("smbshare-controller")
-	}
+	r.recorder = mgr.GetEventRecorderFor("smbshare-controller")
```

Comment:
Just for my education: the linter thinks that recorder is always nil here?

Reply:
The actual output from the linter is: 
```
controllers/smbshare_controller.go:75:19: invalid operation: cannot compare r.recorder == nil (operator == not defined on untyped nil) (typecheck)
```

because  record.EventRecorder is interface

Reply:
Ah, OK.  At first I was a little concerned about this change because it is removing actual logic. But looking at the function being called `setRecorder` its pretty weird that it would have ignored the "set" if recorder was already assigned anyway. I'm now fine with the change.